File: sessions_mount_dir/dlnekhagjohexzh811ixc29mb/frontend/src/api/payments.js
import axios from 'axios';

const API_BASE_URL = '/api/v1';

/**
 * 获取所有支付记录
 * @returns {Promise<Array>} 支付记录列表
 */
export const getPayments = async () => {
  try {
    const response = await axios.get(`${API_BASE_URL}/payments`);
    return response.data;
  } catch (error) {
    console.error('Error fetching payments:', error);
    throw error;
  }
};

/**
 * 获取单个支付记录详情
 * @param {number} paymentId 支付记录ID
 * @returns {Promise<Object>} 支付记录详情
 */
export const getPaymentById = async (paymentId) => {
  try {
    const response = await axios.get(`${API_BASE_URL}/payments/${paymentId}`);
    return response.data;
  } catch (error) {
    console.error(`Error fetching payment with ID ${paymentId}:`, error);
    throw error;
  }
};

/**
 * 创建新的支付记录
 * @param {Object} paymentData 支付数据
 * @param {number} paymentData.member_id 会员ID
 * @param {number} paymentData.amount 支付金额
 * @param {string} paymentData.payment_type 支付类型
 * @returns {Promise<Object>} 新创建的支付记录
 */
export const createPayment = async (paymentData) => {
  try {
    const response = await axios.post(`${API_BASE_URL}/payments`, paymentData);
    return response.data;
  } catch (error) {
    console.error('Error creating payment:', error);
    throw error;
  }
};

/**
 * 获取会员的支付记录
 * @param {number} memberId 会员ID
 * @returns {Promise<Array>} 该会员的支付记录列表
 */
export const getPaymentsByMemberId = async (memberId) => {

/**
 * 更新支付记录
 * @param {number} paymentId 支付记录ID
 * @param {Object} updateData 更新数据
 * @returns {Promise<Object>} 更新后的支付记录
 */
export const updatePayment = async (paymentId, updateData) => {

/**
 * 删除支付记录
 * @param {number} paymentId 支付记录ID
 * @returns {Promise<Object>} 删除操作结果
 */
export const deletePayment = async (paymentId) => {
  try {
    const response = await axios.delete(`${API_BASE_URL}/payments/${paymentId}`);
    return response.data;
  } catch (error) {
    console.error(`Error deleting payment with ID ${paymentId}:`, error);
    throw error;
  }
};
  try {
    const response = await axios.put(`${API_BASE_URL}/payments/${paymentId}`, updateData);
    return response.data;
  } catch (error) {
    console.error(`Error updating payment with ID ${paymentId}:`, error);
    throw error;
  }
};
  try {
    const response = await axios.get(`${API_BASE_URL}/payments`, {
      params: { member_id: memberId }
    });
    return response.data;
  } catch (error) {
    console.error(`Error fetching payments for member ${memberId}:`, error);
    throw error;
  }
};
  try {
    const response = await axios.get(`${API_BASE_URL}/payments`, {
      params: { member_id: memberId }
    });
    return response.data;
  } catch (error) {
    console.error(`Error fetching payments for member ${memberId}:`, error);
    throw error;
  }
};

export default {
  getPayments,
  getPaymentById,
  createPayment,
  getPaymentsByMemberId,
  updatePayment,
  deletePayment
};

File: sessions_mount_dir/cdzcpk0m78430xqx44gjqsbox/backend/app/routers/payments.py
from fastapi import APIRouter, Depends, HTTPException, status
from sqlalchemy.orm import Session
from typing import List

from ..models.payments import Payment, MembershipCard, BillingRecord
from ..schemas.payments import PaymentCreate, PaymentUpdate
from ..database import get_db

router = APIRouter(
    prefix="/api/v1/payments",
    tags=["payments"],
    responses={404: {"description": "Not found"}},
)


@router.post("/", response_model=Payment, status_code=status.HTTP_201_CREATED)
async def create_payment(payment: PaymentCreate, db: Session = Depends(get_db)):
    """
    Create a new payment record
    """
    try:
        db_payment = Payment(**payment)
        db.add(db_payment)
        db.commit()
        db.refresh(db_payment)
        return db_payment
    except Exception as e:
        db.rollback()
        raise HTTPException(
            status_code=status.HTTP_400_BAD_REQUEST,
            detail=f"Error creating payment: {str(e)}"
        )


@router.get("/", response_model=List[Payment])
async def get_payments(skip: int = 0, limit: int = 100, db: Session = Depends(get_db)):
    """
    Get all payment records with pagination
    """
    return db.query(Payment).offset(skip).limit(limit).all()


@router.get("/{payment_id}", response_model=Payment)
async def get_payment(payment_id: int, db: Session = Depends(get_db)):
    """
    Get a specific payment by ID
    """
    payment = db.query(Payment).filter(Payment.id == payment_id).first()
    if not payment:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail=f"Payment with id {payment_id} not found"
        )
    return payment


@router.get("/members/{member_id}", response_model=List[Payment])
async def get_member_payments(member_id: int, db: Session = Depends(get_db)):
    """
    Get all payments for a specific member
    """
    payments = db.query(Payment).filter(Payment.member_id == member_id).all()
    if not payments:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail=f"No payments found for member with id {member_id}"
        )
    return payments


@router.put("/{payment_id}/status", response_model=Payment)
@router.put("/{payment_id}", response_model=Payment)
async def update_payment(payment_id: int, payment: PaymentUpdate, db: Session = Depends(get_db)):
    """
    Update payment details
    """
    db_payment = db.query(Payment).filter(Payment.id == payment_id).first()
    if not db_payment:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail=f"Payment with id {payment_id} not found"
        )

    try:
        update_data = payment.dict(exclude_unset=True)
        for key, value in update_data.items():
            setattr(db_payment, key, value)
        db.commit()
        db.refresh(db_payment)
        return db_payment
    except Exception as e:
        db.rollback()
        raise HTTPException(
            status_code=status.HTTP_400_BAD_REQUEST,
            detail=f"Error updating payment: {str(e)}"
        )
async def update_payment_status(payment_id: int, status_update: dict, db: Session = Depends(get_db)):
    """
    Update payment status
    """
    payment = db.query(Payment).filter(Payment.id == payment_id).first()
    if not payment:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail=f"Payment with id {payment_id} not found"
        )
    
    try:
        payment.status = status_update.get("status", payment.status)
        payment.processed_at = status_update.get("processed_at", payment.processed_at)
        db.commit()
        db.refresh(payment)
        return payment
    except Exception as e:
        db.rollback()
        raise HTTPException(
            status_code=status.HTTP_400_BAD_REQUEST,
            detail=f"Error updating payment status: {str(e)}"
        )


@router.get("/members/{member_id}/cards", response_model=List[MembershipCard])
async def get_member_cards(member_id: int, db: Session = Depends(get_db)):
    """
    Get all membership cards for a specific member
    """
    cards = db.query(MembershipCard).filter(MembershipCard.member_id == member_id).all()
    if not cards:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail=f"No membership cards found for member with id {member_id}"
        )
    return cards


@router.get("/members/{member_id}/billing", response_model=List[BillingRecord])
@router.get("/summary/{member_id}", response_model=dict)
async def get_payment_summary(member_id: int, db: Session = Depends(get_db)):
    """
    Get payment summary for a member including:
    - Total payments
    - Last payment date
    - Outstanding balance
    """
    payments = db.query(Payment).filter(Payment.member_id == member_id).all()
    if not payments:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail=f"No payments found for member with id {member_id}"
        )
    
    total_amount = sum(p.amount for p in payments)
    last_payment = max(p.payment_date for p in payments)
    
    # Calculate outstanding balance from billing records
    billing_records = db.query(BillingRecord).filter(BillingRecord.member_id == member_id).all()
    billed_amount = sum(r.amount for r in billing_records)
    outstanding = billed_amount - total_amount
    
    return {
        "total_payments": len(payments),
        "total_amount": total_amount,
        "last_payment_date": last_payment,
        "outstanding_balance": outstanding if outstanding > 0 else 0
    }
async def get_member_billing_records(member_id: int, db: Session = Depends(get_db)):
    """
    Get all billing records for a specific member
    """
    records = db.query(BillingRecord).filter(BillingRecord.member_id == member_id).all()
    if not records:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail=f"No billing records found for member with id {member_id}"
        )
    return records